VotingApplication/VotingApplication.Web.Api/Controllers/API Controllers/ManageInvitationController.cs

```diff
@@ -0,0 +1,137 @@
+using VotingApplication.Data.Context;
+using VotingApplication.Data.Model;
+
+namespace VotingApplication.Web.Api.Controllers.API_Controllers
+{
+    public class ManageInvitationController : WebApiController
+    {
+        private IMailSender _mailSender;
+
+        public ManageInvitationController() : base()
+        {
+            _mailSender = new MailSender();
+        }
+        public ManageInvitationController(IContextFactory contextFactory, IMailSender mailSender) : base(contextFactory)
+        {
+            _mailSender = mailSender;
+        }
+
+        #region GET
+
+        public HttpResponseMessage Get(Guid manageId)
+        {
+            return this.Request.CreateErrorResponse(HttpStatusCode.MethodNotAllowed, "Cannot use GET on this controller");
+        }
+
+        public HttpResponseMessage Get(Guid manageId, long invitationId)
+        {
+            return this.Request.CreateErrorResponse(HttpStatusCode.MethodNotAllowed, "Cannot use GET by Id on this controller");
+        }
+
+        #endregion
+
+        #region POST
+
+        public HttpResponseMessage Post(Guid manageId, List<string> invitationEmails)
+        {
+            using (var context = _contextFactory.CreateContext())
+            {
+                Poll matchingPoll = context.Polls.Where(p => p.ManageID == manageId).FirstOrDefault();
+                if (matchingPoll == null)
+                {
+                    return this.Request.CreateErrorResponse(HttpStatusCode.NotFound, string.Format("Poll {0} not found", manageId));
+                }
+
+                SendVoteEmails(matchingPoll, invitationEmails);
+
+                return this.Request.CreateResponse(HttpStatusCode.OK);
+            }
+        }
+
+        private void SendVoteEmails(Poll poll, List<string> invitationEmails)
+        {
+            String hostUri = WebConfigurationManager.AppSettings["HostURI"];
+            if (hostUri == String.Empty)
+            {
+                return;
+            }
+
+            foreach (string targetEmailAddress in invitationEmails)
+            {
+                if (String.IsNullOrEmpty(targetEmailAddress))
+                {
+                    continue;
+                }
+
+                //Remove leading/trailing whitespace
+                var emailAddress = targetEmailAddress.Trim();
+                //Check email is correctly formed to avoid costly sending of invalid emails, potentially dropping our SendGrid reputation
+                if (!Regex.IsMatch(emailAddress, @"^[\w._%+-]+@\w+(\.\w+)+$", RegexOptions.IgnoreCase))
+                {
+                    continue;
+                }
+
+
+                string tokenString = "";
+                if (poll.InviteOnly)
+                {
+                    Token token = new Token() { PollId = poll.UUID, TokenGuid = Guid.NewGuid() };
+                    tokenString = "&token=" + token.TokenGuid;
+                    poll.Tokens.Add(token);
+                }
```

Comment:
Just a heads up that this will need changing as per the architectural restructuring.
~/Poll/?poll=<PGUID>&token=<TGUID> Becomes
~/Poll/Index/<PGUID>/<TGUID>
Putting this here as it won't cause any test failures or compile errors, but will still cause bugs we aren't likely to check.